### PR TITLE
test: improve revm-bytecode unittest coverage

### PR DESCRIPTION
**Pull request by JackDrogon**
test: improve revm-bytecode unittest coverage

- Add comprehensive tests for bytecode iterator methods (peek, as_slice, skip_to_next_opcode)
- Add tests for EIP7702 bytecode edge cases and error display
- Add tests for jump table trait implementations (PartialOrd, Hash, Debug)
- Add tests for legacy analyzed bytecode trait methods
- Add tests for opcode helper functions and edge cases

Remaining uncovered lines are assertion failure format strings and unreachable code branches.

### CODE REVIEW

File: crates/bytecode/src/utils.rs [hunk 1]
```diff
@@ -19,3 +19,135 @@ pub unsafe fn read_i16(ptr: *const u8) -> i16 {
 pub unsafe fn read_u16(ptr: *const u8) -> u16 {
     u16::from_be_bytes(unsafe { ptr.cast::<[u8; 2]>().read() })
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_read_u16_big_endian() {
```
Comment (rakita) [inline crates/bytecode/src/utils.rs:28]: For example, this one is the only needed test for `read_u16`.

`test_read_u16_all_ones` is not a valid test in this context.
